sdk/rust/oak_tests/src/lib.rs

```diff
@@ -134,102 +147,50 @@ pub fn runtime_config_wasm(
-        .expect("failed to serialize GrpcRequest message");
-
-    // Create a new channel to hold the request message.
-    //
-    // In most cases we do not care about labels, so we use the least privileged label for this
-    // channel.
-    let (req_write_handle, req_read_handle) = proxy
-        .channel_create(&oak_abi::label::Label::public_untrusted())
-        .expect("could not create channel");
-    proxy
-        .channel_write(req_write_handle, req_msg)
-        .expect("could not write message");
-
-    // Create a new channel for responses to arrive on and also attach that to the message.
-    //
-    // In most cases we do not care about labels, so we use the least privileged label for this
-    // channel.
-    let (rsp_write_handle, rsp_read_handle) = proxy
-        .channel_create(&oak_abi::label::Label::public_untrusted())
-        .expect("could not create channel");
-
-    // Create a notification message and attach the method-invocation specific channels to it.
-    let notify_msg = oak_runtime::NodeMessage {
-        data: vec![],
-        handles: vec![req_read_handle, rsp_write_handle],
-    };
-
-    // Send the notification message (with attached handles) into the Node under test.
-    proxy
-        .channel_write(handle, notify_msg)
-        .expect("could not write message");
-    proxy
-        .channel_close(req_write_handle)
-        .expect("failed to close channel");
-    proxy
-        .channel_close(req_read_handle)
-        .expect("failed to close channel");
-    proxy
-        .channel_close(rsp_write_handle)
-        .expect("failed to close channel");
-
-    // Read the serialized, encapsulated response.
+/// Build a channel and interceptor suitable for building a client that connects
+/// to a Runtime under test.
+pub async fn channel_and_interceptor() -> (Channel, impl Into<tonic::Interceptor>) {
+    // Build a channel that connects to the Runtime under test.  Allow a bit
+    // of leeway in case the server is still starting up.
+    let uri = RUNTIME_URI.parse().expect("Error parsing URI");
+    let tls_config = ClientTlsConfig::new()
+        .ca_certificate(Certificate::from_pem(include_str!("../certs/ca.pem")));
+    let builder = Channel::builder(uri).tls_config(tls_config);
+
+    let mut retries = 0;
+    let channel;
     loop {
-        let rsp = match proxy.channel_read(rsp_read_handle) {
-            Ok(Some(r)) => r,
-            Ok(None) => {
-                debug!("no pending gRPC response message; poll again soon");
-                std::thread::sleep(std::time::Duration::from_millis(100));
-                continue;
+        match builder.connect().await {
+            Ok(c) => {
+                debug!("Connected to gRPC server");
+                channel = c;
+                break;
             }
-            Err(e) => {
-                panic!("failed to read from response channel: {:?}", e);
+            Err(err) => {
+                if retries < RETRY_COUNT {
```

Comment:
Retries are needed because the Runtime is starting up on another thread, and may take a while before it responds to gRPC connections (particularly in the TSAN tests).

(The comment at l.154 is a bit separated in the code, and even more separated in the GitHub diff.  I've extended it and moved it closer to the loop.)